scripts/linters/general_purpose_linter.py

```diff
@@ -744,6 +744,40 @@ def _check_bad_patterns(self):
                 python_utils.PRINT(summary_message)
         return summary_messages
 
+    def _check_newline_at_eof(self):
+        """This function is used to detect newline at the end of file."""
+        if self.verbose_mode_enabled:
+            python_utils.PRINT(
+                'Starting newline at eof check\n'
+                '----------------------------------------')
+        summary_messages = []
+        files_to_lint = self.all_filepaths
+        failed = False
+
+        with linter_utils.redirect_stdout(sys.stdout):
+            for filepath in files_to_lint:
+                file_content = FILE_CACHE.readlines(filepath)
+                if not re.search(r'[^\n]\n', file_content[-1]):
+                    summary_message = (
+                        '%s --> There should be a single newline at the '
+                        'end of file.' % filepath)
+                    summary_messages.append(summary_message)
+                    python_utils.PRINT(summary_message)
+                    failed = True
+
+            if failed:
+                summary_message = (
+                    '%s   Newline at the eof check failed.' % (
+                        _MESSAGE_TYPE_FAILED))
+            else:
+                summary_message = (
+                    '%s   Newline at the eof check passed.' % (
+                        _MESSAGE_TYPE_SUCCESS))
+            summary_messages.append(summary_message)
+            python_utils.PRINT(summary_message)
```

Comment:
As this test will run in multithread, will it be fine to have any random message between the print on line 765 and 777? If no, then print them together once!

Reply:
it will be fine, we have this problem of random messages only with pylint.